### PR TITLE
Switch back to Swift tuples for typed captures

### PR DESCRIPTION
**Pull request by rxwei**
This depends on #114 and #126. Please review 3db17b5.

### CODE REVIEW

File: Sources/Exercises/Participants/RegexParticipant.swift [hunk 1]
```diff
@@ -80,18 +80,18 @@ private func graphemeBreakPropertyData(
...
     }
-    OneOrMore(.whitespace)
+    oneOrMore(.whitespace)
     ";"
-    OneOrMore(.whitespace)
-    OneOrMore(.word).tryCapture(Unicode.GraphemeBreakProperty.init)
-    Repeat(.any)
+    oneOrMore(.whitespace)
+    oneOrMore(.word).tryCapture(Unicode.GraphemeBreakProperty.init)
+    many(.any)
```
[the remark this answers is not in the document]
Comment (rxwei) [reply Sources/Exercises/Participants/RegexParticipant.swift:92]: This is from another PR (#126). Let's discuss this there.

**Comment by rxwei** [comment]
@swift-ci please test linux

File: Sources/_StringProcessing/RegexDSL/Core.swift [hunk 2]
```diff
@@ -167,10 +145,8 @@ extension RegexProtocol {
         return nil
       }
       let convertedMatch: Match
-      if Match.self == Tuple2<Substring, DynamicCaptures>.self {
-        convertedMatch = Tuple2(
-          input[range], DynamicCaptures(captures)
-        ) as! Match
+      if Match.self == (Substring, DynamicCaptures).self {
+        convertedMatch = (input[range], DynamicCaptures(captures)) as! Match
```
Comment (milseman) [inline Sources/_StringProcessing/RegexDSL/Core.swift:149]: I was wondering, what is this code path for?

**Review by milseman — APPROVED** [review]
LGTM, if I'm reading the diffs right between this and #126

File: Sources/_StringProcessing/RegexDSL/Core.swift (hunk 2 above)
Comment (rxwei) [reply Sources/_StringProcessing/RegexDSL/Core.swift:149]: See `testDynamicCaptures()`. This is used for compiling and running a regex string that's not known at compile time (https://github.com/apple/swift-experimental-string-processing/pull/36).

**Comment by rxwei** [comment]
@swift-ci please test Linux

**Pull request merged**